pkg/imgpkg/signature/cosign.go

```diff
@@ -0,0 +1,52 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package signature
+
+import (
+	"fmt"
+	"net/http"
+
+	regname "github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/remote/transport"
+	"github.com/k14s/imgpkg/pkg/imgpkg/imageset"
+	"github.com/k14s/imgpkg/pkg/imgpkg/registry"
+	"github.com/k14s/imgpkg/pkg/imgpkg/signature/cosign"
+)
+
+func NewCosign(reg registry.Registry) *Cosign {
+	return &Cosign{registry: reg}
+}
+
+type Cosign struct {
+	registry registry.Registry
+}
+
+func (c Cosign) Signature(imageRef regname.Digest) (imageset.UnprocessedImageRef, error) {
+	sigTagRef, err := c.signatureTag(imageRef)
+	if err != nil {
+		return imageset.UnprocessedImageRef{}, err
+	}
+
+	signature, err := c.registry.Digest(sigTagRef)
```

Comment:
may be better: signature -> sigDigestRef?

Reply:
done